sched/scheduler/stateful_scheduler.go

```diff
@@ -270,7 +265,8 @@ func (s *statefulScheduler) scheduleTasks() {
 	}
 
 	// Calculate a list of Tasks to Node Assignments & start running all those jobs
-	taskAssignments := getTaskAssignments(s.clusterState, unscheduledTasks)
+	taskAssignments, affinity := getTaskAssignments(s.clusterState, unscheduledTasks)
```

Comment:
if we make getTaskAssignments into getTaskAssignment and move into the scheduling loop below (make that loop, loop over unscheduledTasks), I think we can eliminate the logic for copying the nodeSet in getTaskAssignment.  getTaskAssignment can just look for an idle node with that snapshotId, otherwise and idle node with snapshotID "", otherwise, search through all of clusterState's nodesets for the first Idle node.